utils/joiner/joinpartition.cpp

```diff
@@ -122,6 +119,11 @@ JoinPartition::JoinPartition(const RowGroup& lRG,
     {
         compressor.reset(new compress::CompressInterfaceSnappy());
     }
+
+    for (int i = 0; i < (int) bucketCount; i++)
```

Comment:
Could you share the semantics of this particular change? Why did you move this block? And if you are going to touch this could you remove an explicit int cast?

Reply:
This is the change for  MCOL-4846 Switch to disk based join causes segv.
The call to "new JoinPartition(*this, false)" attempts to copy the compressor, but in the code's old position, compressor hadn't been set yet. This potentially caused segv. I just moved the code. I can remove the (int).